libs/common/utilities/src/lib/util/utils.ts

```diff
@@ -146,3 +146,14 @@ export const returnNotEmptyString = (s: string) => (s && s !== '' ? s : undefine
 export const returnNumber = (n: number) => (!isNaN(n) ? n : undefined);
 export const ifDefinedString = (s: string) => ifDefined(returnNotEmptyString(s));
 export const ifDefinedNumber = (n: number) => ifDefined(returnNumber(n));
+
+export const findDeepestElementThroughShadowRoot = (rootElement: Element | ShadowRoot | null, selector: string): Element | null => {
```

Comment:
is wel duidelijk wat het doet maar mss wel tof als je er nog lijntje documentatie bijzet wat het doet? je mag naar eigen inzicht kiezen of je dit nog erbij zet